lang/c++/impl/Node.cc

```diff
@@ -146,7 +146,7 @@ void Node::setLogicalType(LogicalType logicalType) {
             if (type_ == AVRO_FIXED) {
                 // Max precision that can be supported by the current size of
                 // the FIXED type.
-                long maxPrecision = floor(log10(2.0) * (8.0 * fixedSize() - 1));
+                int32_t maxPrecision = static_cast<int32_t>(floor(log10(2.0) * (8.0 * static_cast<double>(fixedSize()) - 1)));
```

Comment:
@Fokko I didn't introduce this here, so I don't think I should fix it here... but I believe this calculation is incorrect.

According to [the spec](https://avro.apache.org/docs/1.11.1/specification/#decimal), the maximum precision should be:

$$
floor(log_{10}(2^{8 \times n - 1} - 1))
$$

Rust is the only language implementation that appears to implement this correctly. Python, Java, and C++ all implement this as:

$$
floor(log_{10}(2^{8 \times n - 1}))
$$

JavaScript appears to have an example of a `DecimalType` with this same issue, but it doesn't appear to be in any library code shipped by the library.